presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -153,4 +164,21 @@ public static HiveColumnHandle updateRowIdHandle(String connectorId)
         // the plan-time support needed just works.
         return new HiveColumnHandle(connectorId, UPDATE_ROW_ID_COLUMN_NAME, HIVE_LONG, BIGINT.getTypeSignature(), -1, true);
     }
+
+    public static HiveColumnHandle pathColumnHandle(String connectorId)
+    {
+        // Like partition columns, this column can not be found in data source. Using true for partitionKey,
+        // the plan-time support needed just works.
+        return new HiveColumnHandle(connectorId, PATH_COLUMN_NAME, PATH_HIVE_TYPE, PATH_TYPE_SIGNATURE, PATH_COLUMN_INDEX, true);
```

Comment:
Considering hidden columns as a partition key is quick and dirty hack. Such hacks are only acceptable when we are in desperate need of a feature or bug fix. Most of time, we avoid doing this.

Instead, change `boolean partitionKey` into `columnType` of enum type.